lib/sql-string.js

```diff
@@ -61,6 +61,10 @@ SqlString.escape = function(val, stringifyObjects, timeZone, dialect, field) {
     return val + '';
   }
 
+  if (field && field.type && field.type.validate) {
+    field.type.validate(val);
```

Comment:
@fixe I'm still giving it some thought, i like the idea of catching all types before they hit the database but `escape` seems like an odd place to have it (but yes it's likely the only place being called by all types of values before going into sql).

@janmeier ?